ui/e2e/src/tests/timeSeriesChartLegends.spec.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import happoPlaywright from 'happo-playwright';
+import { mockTimeSeriesResponseWithStableValue } from '@perses-dev/internal-utils';
+import { test } from '../fixtures/dashboardTest';
+import { waitForStableCanvas } from '../utils';
+
+test.use({
+  dashboardName: 'TimeSeriesChartLegends',
+  mockNow: 1673805600000,
+});
+
+test.describe('Dashboard: Time Series Chart Legends', () => {
+  test.beforeEach(async ({ context }) => {
+    await happoPlaywright.init(context);
+  });
+
+  test.afterEach(async () => {
+    await happoPlaywright.finish();
+  });
+
+  ['bottom table', 'right table', 'bottom list', 'right list'].forEach((panelName) => {
+    test(`on mouse over ${panelName} legend item highlights the associated line`, async ({
```

Comment:
Testing the line chart + legend hover behavior in this e2e test because testing the imperative handle + echarts combo in `LineChart` with RTL would be challenging.